### PR TITLE
Add first entry in ChangeLog for v0.1.0 release

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
First entry in ChangeLog, "introducing UMF to the world" 😉 

I intend to make this commit a tag commit, so probably won't merge this as PR, but push it directly to the main.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,7 +1,24 @@
-<day of week> <month> <day> 2024 <Author> <author@intel.com>
+<day of week> May <day> 2024 <Author> <author@intel.com>
```
Comment (bratpiorka) [inline ChangeLog:1]: why you fill only month?

File: ChangeLog [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+	attributes, allowing certain allocation types to be isolated from others and
+	allocated using different hardware resources as required.
+
+	Documentation of UMF, including API description, architecture, and examples
+	can be found: https://oneapi-src.github.io/unified-memory-framework
+
+	Features:
+	- Level Zero memory provider
+	- OS memory provider
+	- jemalloc pool (based on jemalloc)
```
Comment (KFilipek) [inline ChangeLog:19]: `- Jemalloc pool (based on jemalloc)`

File: ChangeLog (hunk 1 above)
Comment (lukaszstolarczuk) [reply ChangeLog]: I wanted to make sure I update it with the proper date - anyway I added it already.

File: ChangeLog (hunk 2 above)
Comment (lukaszstolarczuk) [reply ChangeLog]: Done.

**Pull request closed without merging**